ownCloud/Client/Actions/Actions+Extensions/DuplicateAction.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  DuplicateAction.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 15/11/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+import Foundation
+
+import ownCloudSDK
+
+class DuplicateAction : Action {
+	override class var identifier : OCExtensionIdentifier? { return OCExtensionIdentifier("com.owncloud.action.duplicate") }
+	override class var category : ActionCategory? { return .normal }
+	override class var name : String? { return "Duplicate".localized }
+
+	// MARK: - Extension matching
+	override class func applicablePosition(forContext: ActionContext) -> ActionPosition {
+		// Examine items in context
+		return .middle
+	}
+
+	// MARK: - Action implementation
+	override func run() {
+		guard context.items.count > 0 else {
+			completionHandler?(NSError(ocError: .errorInsufficientParameters))
+			return
+		}
+
+		beforeRunHandler?()
+
+		let item = context.items[0]
+		let rootItem = context.items[1]
```

Comment:
`items` should only contain items to perform the action on, and might end up having different parent items. Therefore, the `parentItem` should be retrieved from `OCCore` dynamically.

I wrote an OCItem category that does this for you:
```swift
extension OCItem {
	func parentItem(from core: OCCore, completionHandler: ((_ error: Error?, _ parentItem: OCItem?) -> Void)? = nil) -> OCItem? {
		var parentItem : OCItem?

		if let parentItemIdentifier = self.parentFileID {
			var waitGroup : DispatchGroup?

			if completionHandler == nil {
				waitGroup = DispatchGroup()
				waitGroup?.enter()
			}

			core.retrieveItemFromDatabase(forFileID: parentItemIdentifier) { (error, _, item) in
				if completionHandler == nil {
					parentItem = item
					waitGroup?.leave()
				} else {
					completionHandler?(error, item)
				}
			}

			waitGroup?.wait()
		}

		return parentItem
	}
}
```

It can be used both synchronous and asynchronously:
```swift
 // Synchronous
let parentItem = item.parentItem(from: core)

 // Asynchronous (and preferred)
item.parentItem(from: core) { (_ error, _ parenItem) -> Void in 
    OnMainThread {
        // work with parentItem
    }
}
```